packages/redux-dynostore-core/src/mergeReducers.js

```diff
@@ -21,7 +23,7 @@ const removeUndefinedValues = (state = {}) => {
 const mergeReducers = reducers => (state, action) => {
   const nextState = reducers.reduce((nextState, reducer) => {
     const newState = reducer(state !== undefined ? nextState : undefined, action)
-    return newState !== nextState ? { ...nextState, ...newState } : nextState
+    return newState !== nextState ? deepMerge(nextState, newState) : nextState
```

Comment:
I thought about this a little more. The deep merge is actually the behaviour you would expect, and should be the default (making a major bump). The shallow should probably be added in the same release so that anyone adversely affected by the performance changes can use the old method.

Reply:
Do you think a flag or allowing them to override the whole merge function is better? 

I don't think we would ship a `shallowMerge` function as it's fairly easy to write (`(a, b) => ({ ...a, ...b })`).  Thoughts?

Reply:
I like the idea of providing a merge function, allowing consumers to specify a different way of dealing with things... I think we should export the shallowMerge function still, its tiny and would probably the most useful override for anyone.